From 77f6848ad5962dfa809f765d2bf26bc458b7fa22 Mon Sep 17 00:00:00 2001
From: Frank Wang <frank.wang@rock-chips.com>
Date: Fri, 16 Dec 2016 09:09:23 +0800
Subject: [PATCH] phy: rockchip-inno-usb2: rm phy_power_on/off in
 suspend/resume cases.

Usb-controller can invoke phy_power_on/off in its suspend/resume
process, so usb-phy need not do it again.

This adds remove phy_power_on/off in its suspend/resume cases.

Change-Id: Ice30e79ffba8116ca9bfae344c7ea232f6580130
Signed-off-by: Frank Wang <frank.wang@rock-chips.com>
---
 drivers/phy/phy-rockchip-inno-usb2.c | 25 -------------------------
 1 file changed, 25 deletions(-)

diff --git a/drivers/phy/phy-rockchip-inno-usb2.c b/drivers/phy/phy-rockchip-inno-usb2.c
index b5321bac9e12..5f8597ab0fa1 100644
--- a/drivers/phy/phy-rockchip-inno-usb2.c
+++ b/drivers/phy/phy-rockchip-inno-usb2.c
@@ -1356,13 +1356,6 @@ static int rockchip_usb2phy_pm_suspend(struct device *dev)
 		if (!rport->phy)
 			continue;
 
-		if (!rport->suspended) {
-			if (rport->port_id == USB2PHY_PORT_HOST)
-				rockchip_usb2phy_sm_work(&rport->sm_work.work);
-			else
-				rockchip_usb2phy_power_off(rport->phy);
-		}
-
 		/* activate the linestate to detect the next interrupt. */
 		property_enable(rphy, &rport->port_cfg->ls_det_clr, true);
 		property_enable(rphy, &rport->port_cfg->ls_det_en, true);
@@ -1373,24 +1366,6 @@ static int rockchip_usb2phy_pm_suspend(struct device *dev)
 
 static int rockchip_usb2phy_pm_resume(struct device *dev)
 {
-	struct rockchip_usb2phy *rphy = dev_get_drvdata(dev);
-	struct rockchip_usb2phy_port *rport;
-	int index;
-
-	for (index = 0; index < rphy->phy_cfg->num_ports; index++) {
-		rport = &rphy->ports[index];
-		if (!rport->phy)
-			continue;
-
-		/*
-		 * Resuming case, for host-port, *_linestate_irq() will take
-		 * over all actions, but for otg-port, we should invoke
-		 * *_power_on() to resume the phy-port manually.
-		 */
-		if (rport->suspended && rport->port_id == USB2PHY_PORT_OTG)
-			rockchip_usb2phy_power_on(rport->phy);
-	}
-
 	return 0;
 }
 
-- 
2.35.3

